api/views/stories.py

```diff
@@ -24,31 +25,51 @@ def get_stories():
 def post_stories():
     data = request.get_json()
     if 'story_name' in data:
-        new_story = row_constructor(Story, data)
+        if data['story_name'] is not None:
+            new_story = row_constructor(Story, data)
+        else:
+            return create_response(status=422, message='The story_name parameter was invalid')
     else:
         return create_response(status=422, message='No story_name parameter was provided')
     db.session.add(new_story)
     db.session.flush()
 
     if 'poi_ids' in data:
-        new_story_pois = [row_constructor(StoryPOI, story_id=new_story.id, poi_id=poi_id) for poi_id in data['poi_ids']]
-        db.session.add_all(new_story_pois)
+        if data['poi_ids'] is not None:
+            new_story_pois = [row_constructor(StoryPOI, story_id=new_story.id, poi_id=poi_id) for poi_id in data['poi_ids']]
+            db.session.add_all(new_story_pois)
     db.session.commit()
     story = Story.query.get(new_story.id)
     return create_response(data={'story': story.to_dict()}, message='Story created')
 
 @app.route(STORIES_ID_URL, methods=['PUT'])
 def put_stories(story_id):
     data = request.get_json()
+    fields = ['story_name', 'poi_ids']
+    missing_params = [field for field in fields if data.get(field) is None]
+
     story = Story.query.get(story_id)
     if 'story_name' in data:
-        story.story_name = data['story_name']
-    if 'poi_ids' in data:
-        StoryPOI.query.filter(StoryPOI.story_id == story_id).delete()
-        if(len(data['poi_ids']) > 0):
-            new_story_pois = [row_constructor(StoryPOI, story_id=story_id, poi_id=poi_id) for poi_id in data['poi_ids']]
-            db.session.add_all(new_story_pois)
+        if data['story_name'] is not None:
+            story.story_name = data['story_name']
             db.session.commit()
+        else:
+            missing_params.append('story_name')
+
+    if 'poi_ids' in data:
+        if data['poi_ids'] is not None:
+            StoryPOI.query.filter(StoryPOI.story_id == story_id).delete()
+            if(len(data['poi_ids']) > 0):
+                new_story_pois = [row_constructor(StoryPOI, story_id=story_id, poi_id=poi_id) for poi_id in data['poi_ids']]
+                db.session.add_all(new_story_pois)
+                db.session.commit()
+        else:
+            missing_params.append('poi_ids')
+
+    if len(missing_params) == 2:

```

Comment:
you should move this after the `missing_params = ...` line. typically its better to handle your error cases early, kinda as a way to shield the "valid" code for valid inputs if that makes any sense